core/src/avm2/globals/math.rs

```diff
@@ -107,20 +110,44 @@ pub fn min<'gc>(
         let val = arg.coerce_to_number(activation)?;
         if val.is_nan() {
             return Ok(f64::NAN.into());
-        } else if val < cur_min {
+        } else if val < cur_min
+            || (val == cur_min && val.is_sign_negative() && !cur_min.is_sign_negative())
+        {
             cur_min = val;
         }
     }
     Ok(cur_min.into())
 }
 
 pub fn pow<'gc>(
-    _activation: &mut Activation<'_, 'gc>,
+    activation: &mut Activation<'_, 'gc>,
     _this: Value<'gc>,
     args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
-    let n = args[0].as_f64();
-    let p = args[1].as_f64();
+    let n = args[0].coerce_to_number(activation)?;
+    let p = args[1].coerce_to_number(activation)?;
+
+    // If y is NaN, the result is NaN.
```

Comment:
Comments refer to values as `x` and `y` whereas code refers to them as `n` and `p`. This should be unified somehow (it's better to use the names from docs if possible I guess).